client/client_test.go

```diff
@@ -1253,15 +1249,9 @@ func assertPublishToRolesSucceeds(t *testing.T, repo1 *NotaryRepository,
 	assert.NoError(t, err)
 	assert.Len(t, getChanges(t, repo1), 0, "wrong number of changelist files found")
 
-	// Create a new repo and pull from the server
-	tempBaseDir, err := ioutil.TempDir("", "notary-test-")
-	defer os.RemoveAll(tempBaseDir)
-
-	assert.NoError(t, err, "failed to create a temporary directory: %s", err)
-
-	repo2, err := NewNotaryRepository(tempBaseDir, repo1.gun, repo1.baseURL,
-		http.DefaultTransport, passphraseRetriever)
-	assert.NoError(t, err, "error creating repository: %s", err)
+	// use another repo to check metadata
+	repo2 := newRepoToTestRepo(t, repo1)
```

Comment:
I check that both the original repo and the new clean repo have the expected roles (in the for loop)